File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/view/UserListView.java
package io.jeffchang.buttondemo.ui.userlist.view;

import java.util.List;

import io.jeffchang.buttondemo.models.User;

/**
 * View contract for the list fragment.
 */

public interface UserListView {

    void onGetUsersSuccess(List<User> user);

    void onGetUsersFailure(Throwable throwable);
}

File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/view/UserListView.kt
package io.jeffchang.buttondemo.ui.userlist.view

import io.jeffchang.buttondemo.models.User

/**
 * View contract for the list fragment.
 */

interface UserListView {

    fun onGetUsersSuccess(users: List<User>)

    fun onGetUsersFailure(throwable: Throwable)
}